src/Io/StreamingServer.php

```diff
@@ -96,19 +97,19 @@ final class StreamingServer extends EventEmitter
      * connections in order to then parse incoming data as HTTP.
      * See also [listen()](#listen) for more details.
      *
-     * @param LoopInterface $loop
      * @param callable $requestHandler
+     * @param float $idleConnectTimeout
      * @see self::listen()
      */
-    public function __construct(LoopInterface $loop, $requestHandler)
+    public function __construct($requestHandler, RequestHeaderParser $parser, Clock $clock)
```

Comment:
This change makes the diff quite massive, I wonder if that's something we can reasonably avoid?

Reply:
What do you have in mind? Regardless of what we do it will be a massive diff. What I could do is file another PR updating the tests moving some things around that will make the diff in this PR a lot less.

Reply:
Just filed https://github.com/reactphp/http/pull/469 preparing to keep the change set smaller here,